server/src/main/java/org/elasticsearch/cluster/service/MasterService.java

```diff
@@ -74,6 +78,8 @@ public class MasterService extends AbstractLifecycleComponent {
 
     static final String MASTER_UPDATE_THREAD_NAME = "masterService#updateTask";
 
+    public static final String STATE_UPDATE_ACTION_NAME = "internal:cluster/coordination/update_state";
```

Comment:
This looks like a transport action name. In other places where we register a task, we pick a simpler name with no "internal:" prefix, for instance for enrich, it is just `policy_execution`. I'd prefer the same here to avoid the confusion. `publish_cluster_state_update` for instance.

Reply:
But there are also examples in the same form - for example, `JoinHelper` has 3 action names that all start with `internal:cluster/coordination`. The `internal:` prefix will also make it easier to filter out cluster management tasks when capturing traces, because we can filter in and filter out by span name, which here equates to task name.

Reply:
Those 3 are different, they are all 3 registered as transport actions and used on the wire. A common prefix for all artificial actions would be fine though, we just have not had it. We could start one now or leave this without one. I'd prefer not to reuse the space we have. I think filtering `internal:` is not going to be useful anyway, so many actions under that anyway.

Reply:
I agree that filtering on prefixes like `internal:` is probably not helpful, and indeed end-users probably don't know much about the structure of these action names. It would have been good to have all action names following the same structure as the ones for transport actions, but I think we've already crossed that bridge unfortunately. So yes a bare `publish_cluster_state_update` would be ok with me.

Reply:
I updated the task name, and had a go at adding a test in `MasterServiceTests`. Please let me know what you think, I hacked it together from other examples.

I looked into adding something similar for recoveries, but I'm a bit lost. I looked at `PeerRecoverySourceServiceTests` since it's already being changed to create a task to pass in, but the existing test is very narrow in scope and it's not obvious how to problem the task-executing behaviour without lots of new setup code or by opening up method visibility, neither of which is appealing. I'm hoping someone can point somewhere and say "oh, you just need something a bit like that."

...anyone?

Reply:
I had something like `IndexRecoveryIT.testOngoingRecoveryAndMasterFailOver` in mind, i.e., ensure a recovery, then block it through transport, then check that the task is registered correctly (not doing the master failover part of the test).

I'd be happy to hack something together.

Did not look at your test yet, might be sufficient.